### PR TITLE
Functional tests - Add new test to check 'BO > Products page > Virtual tab'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add new test to check 'BO > Products page > Virtual tab'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | no need
| UI Tests          | Nightly => https://github.com/nesrineabdmouleh/testing_pr/actions/runs/6235777770
| Fixed issue or discussion?     | no
| Related PRs       | no
| Sponsor company   |

### CODE REVIEW

File: tests/UI/campaigns/productV2/functional/13_virtualTab.ts [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+import checkoutPage from '@pages/FO/checkout';
+import Customers from '@data/demo/customers';
+import PaymentMethods from '@data/demo/paymentMethods';
+import orderConfirmationPage from '@pages/FO/checkout/orderConfirmation';
+import {homePage} from '@pages/FO/home';
+import {myAccountPage} from '@pages/FO/myAccount';
+import foOrderHistoryPage from '@pages/FO/myAccount/orderHistory';
+import orderDetailsPage from '@pages/FO/myAccount/orderDetails';
+import virtualProductTab from '@pages/BO/catalog/productsV2/add/virtualProductTab';
+import productSettings from '@pages/BO/shopParameters/productSettings';
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/13_virtualTab.ts]:
```suggestion
// Import BO pages
import dashboardPage from '@pages/BO/dashboard';
import createProductsPage from '@pages/BO/catalog/productsV2/add';
import virtualProductTab from '@pages/BO/catalog/productsV2/add/virtualProductTab';
import productsPage from '@pages/BO/catalog/productsV2';
import productSettings from '@pages/BO/shopParameters/productSettings';
// Import FO pages
import {cartPage} from '@pages/FO/cart';
import checkoutPage from '@pages/FO/checkout';
import orderConfirmationPage from '@pages/FO/checkout/orderConfirmation';
import {homePage} from '@pages/FO/home';
import {myAccountPage} from '@pages/FO/myAccount';
import orderDetailsPage from '@pages/FO/myAccount/orderDetails';
import foOrderHistoryPage from '@pages/FO/myAccount/orderHistory';
import foProductPage from '@pages/FO/product';
```

File: tests/UI/campaigns/productV2/functional/13_virtualTab.ts [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+
+// Import data
+import ProductData from '@data/faker/product';
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/13_virtualTab.ts:28]:
```suggestion
import ProductData from '@data/faker/product';
import Customers from '@data/demo/customers';
import PaymentMethods from '@data/demo/paymentMethods';
```

File: tests/UI/campaigns/productV2/functional/13_virtualTab.ts [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+      expect(pageTitle).to.equal(orderDetailsPage.pageTitle);
+    });
+
+    it('should download the file', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkDownloadFile', baseContext);
+
+      await orderDetailsPage.clickOnDownloadLink(page);
+
+      const doesFileExist = await files.doesFileExist(newProductData.fileName, 5000);
+      await expect(doesFileExist, 'File is not downloaded!').to.be.true;
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/13_virtualTab.ts]:
```suggestion
      expect(doesFileExist, 'File is not downloaded!').to.be.eq(true);
```

File: tests/UI/campaigns/productV2/functional/13_virtualTab.ts [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+      });
+
+      it('should go back to BO', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', `goBackToBO${index}`, baseContext);
+
+        // Go back to BO
+        page = await foProductPage.closePage(browserContext, page, 0);
+
+        const pageTitle = await createProductsPage.getPageTitle(page);
+        await expect(pageTitle).to.contains(createProductsPage.pageTitle);
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/13_virtualTab.ts]:
```suggestion
        expect(pageTitle).to.contains(createProductsPage.pageTitle);
```

File: tests/UI/campaigns/productV2/functional/13_virtualTab.ts [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+
+      const pageTitle = await foProductPage.getPageTitle(page);
+      expect(pageTitle).to.contains(newProductData.name);
+    });
+
+    it('should check the product availability label', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkProductAvailableLabel', baseContext);
+
+      const availabilityLabel = await foProductPage.getProductAvailabilityLabel(page);
+      await expect(availabilityLabel).to.contains('Product available');
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/13_virtualTab.ts]:
```suggestion
      expect(availabilityLabel).to.contains('Product available');
```

File: tests/UI/campaigns/productV2/functional/13_virtualTab.ts [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+    });
+
+    it('should check the label of out of stock', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkLabelOutOfStock', baseContext);
+
+      const availabilityLabel = await foProductPage.getProductAvailabilityLabel(page);
+      await expect(availabilityLabel).to.contains('Out of stock');
+
+      const isVisible = await foProductPage.isAddToCartButtonEnabled(page);
+      await expect(isVisible).to.be.true;
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/13_virtualTab.ts]:
```suggestion
      expect(isVisible).to.be.eq(true);
```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

**Review by Progi1984 — DISMISSED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @nesrineabdmouleh

**Pull request merged**